engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -1066,29 +1104,182 @@ public int hashCode()
   @Override
   public <T extends Operator> T addOperator(String name, T operator)
   {
+    name = moduleStack.empty() ? name : moduleStack.peek().getName() + "_" + name;
     if (operators.containsKey(name)) {
-      if (operators.get(name) == (Object)operator) {
+      if (operators.get(name).operator == operator) {
         return operator;
       }
       throw new IllegalArgumentException("duplicate operator id: " + operators.get(name));
     }
 
     OperatorMeta decl = new OperatorMeta(name, operator);
+    decl.setParentModuleName(moduleStack.empty() ? null : moduleStack.peek().getName());
     rootOperators.add(decl); // will be removed when a sink is added to an input port for this operator
     operators.put(name, decl);
     return operator;
   }
 
-  @Override
-  public <T extends Module> T addModule(String name, Class<T> moduleClass)
+  public final class ModuleMeta implements DAG.ModuleMeta, Serializable
   {
-    throw new UnsupportedOperationException("Modules are not supported");
+    private final LinkedHashMap<InputPortMeta, StreamMeta> inputStreams = new LinkedHashMap<InputPortMeta, StreamMeta>();
+    private final LinkedHashMap<OutputPortMeta, StreamMeta> outputStreams = new LinkedHashMap<OutputPortMeta, StreamMeta>();
+    private final Attribute.AttributeMap attributes;
+    @SuppressWarnings("unused")
+    private final int id;
+    @NotNull
+    private final String name;
+    private transient Integer nindex; // for cycle detection
+    private transient Integer lowlink; // for cycle detection
+    private transient Module module;
+    private String parentModuleName;
```

Comment:
Can you add documentation what it refers to which makes it clear why is it meant to be parent module name only?